edx_sga/tests/test_sga.py

```diff
@@ -0,0 +1,617 @@
+from opaque_keys.edx.locator import CourseLocator  # lint-amnesty, pylint: disable=import-error
+from xblock.field_data import DictFieldData
+from xblock.fields import DateTime
+
+from edx_sga.tests.common import DummyResource, DummyUpload
+
+
+SHA1 = 'da39a3ee5e6b4b0d3255bfef95601890afd80709'
+
+
+def fake_get_submission(upload):
+    """retrurns fake submission"""
+    return {
+        "answer": {
+            "sha1": SHA1,
+            "filename": upload.file.name.encode('utf-8'),
+            "mimetype": mimetypes.guess_type(upload.file.name.encode('utf-8'))[0]
+        }
+    }
+
+
+class MockedStudentModule(object):
+    """dummy representation of xblock class"""
+    def __init__(self):
+        self.course_id = CourseLocator(org='foo', course='baz', run='bar')
+        self.module_state_key = "foo"
+        self.student = mock.Mock(username="fred6", is_staff=False, password="test")
+        self.state = '{"display_name": "Staff Graded Assignment"}'
+
+    def save(self):
+        """save method do nothing"""
+        pass
+
+
+@ddt
+class StaffGradedAssignmentMockedTests(unittest.TestCase):
+    """
+    Create a SGA block with mock data.
+    """
+    def setUp(self):
+        """
+        Creates a test course ID, mocks the runtime, and creates a fake storage
+        engine for use in all tests
+        """
+        super(StaffGradedAssignmentMockedTests, self).setUp()
+        self.course_id = CourseLocator(org='foo', course='baz', run='bar')
+        self.runtime = mock.Mock(anonymous_student_id='MOCK')
+        self.scope_ids = mock.Mock()
+        tmp = tempfile.mkdtemp()
+        patcher = mock.patch(
+            "edx_sga.sga.default_storage",
+            FileSystemStorage(tmp))
+        patcher.start()
+        self.addCleanup(patcher.stop)
+        self.staff = mock.Mock(return_value={
+            "password": "test",
+            "username": "tester",
+            "is_staff": True
+        })
+
+    def make_one(self, display_name=None, **kw):
+        """
+        Creates a XBlock SGA for testing purpose.
+        """
+        from edx_sga.sga import StaffGradedAssignmentXBlock as cls
+        field_data = DictFieldData(kw)
+        block = cls(self.runtime, field_data, self.scope_ids)
+        block.location = Location(
+            'foo', 'bar', 'baz', 'category', 'name', 'revision'
+        )
+
+        block.xmodule_runtime = self.runtime
+        block.course_id = self.course_id
+        block.scope_ids.usage_id = "i4x://foo/bar/category/name"
+        block.category = 'problem'
+
+        if display_name:
+            block.display_name = display_name
+
+        block.start = datetime.datetime(2010, 5, 12, 2, 42, tzinfo=pytz.utc)
+        return block
+
+    def test_ctor(self):
+        """
+        Test points are set correctly.
+        """
+        block = self.make_one(points=10)
+        self.assertEqual(block.display_name, "Staff Graded Assignment")
```

Comment:
Can you change these `self.assertEqual` to `assert block.display_name == "Staff Graded Assignment"`, here and below?